### PR TITLE
Finetune the 16GB and 32GB profiles sizes

### PR DESCRIPTION
**Pull request by dmathieu**
## Motivation/summary

I have been finetuning the ES size and number of shards used for those benchmark profiles in the appropriate provider branches. This brings the tuning back into main.

## Checklist

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

Run a benchmark.

### CODE REVIEW

File: testing/benchmark/system-profiles/32GBx2zone.tfvars [hunk 1]
```diff
@@ -3,10 +3,14 @@ user_name = "USER"
...
-elasticsearch_size = "128g"
+elasticsearch_size = "256g"
 # The number of AZs the Elasticsearch cluster should have.
 elasticsearch_zone_count = 2
+# Run the cluster with a dedicated master
+elasticsearch_dedicated_masters = true
 # APM server instance size
 apm_server_size = "32g"
+# Number of shards for the ES indices
+apm_shards = 4
```
Comment (simitt) [inline testing/benchmark/system-profiles/32GBx2zone.tfvars:14]: Is there a specific reason this wasn't set to 8? (1 per 64GB => 256: 4x2)

Comment (dmathieu) [reply testing/benchmark/system-profiles/32GBx2zone.tfvars:14]: The benchmark results are the same whether we go to 4 or 8 shards for 32GB, or 2 to 4 with 16GB.

File: testing/benchmark/system-profiles/16GBx2zone.tfvars [hunk 2]
```diff
@@ -3,10 +3,12 @@ user_name = "USER"
 # The number of AZs the APM Server should span.
 apm_server_zone_count = 1
 # The Elasticsearch cluster node size.
-elasticsearch_size = "64g"
+elasticsearch_size = "128g"
```
Comment (simitt) [inline testing/benchmark/system-profiles/16GBx2zone.tfvars:6]: Have you also tried with an even larger ES instance size + doubling the apm-shards? 
`256GB`, `8` shards?

Comment (dmathieu) [reply testing/benchmark/system-profiles/16GBx2zone.tfvars:6]: I have, for 16GB, and there's no difference.

**Review by marclop — APPROVED** [review]
Thank you!

**Pull request merged**